questions 1-14

```diff
@@ -0,0 +1,96 @@
+1. models.py contains classes and fields where tables represent tables and fields represents the column types.
+   Some field options are null,  blank, default, editable unique and more.
+   models.py is mostly incharge of containing the DBs build state.
+
+2. Views is incharge of the logic in the backend. That's where all of the functionality is done for the clients requests.
+   The functions in the view can retrieve data from the db, send an email and do lots of more backend related logic.
+
+3. The url.py file holds all of the URLs that our project works with.
+   When a client enters a url it checks the url.py and if it exists it then will send the request to a view function
+   that will handle the clients request.
+
+4. Migration is the process of applying changes to your model (db schema). Its incharge of keeping your model and db in sync.
+   To add a new migration you need to write into the cmd line: manage.py migrate.
+
+5. The serializer converts complex data types like django model instances into python objects so that we can then
+   convert it into json. The reason we do this is because some things are to complex to immediately convert into json format
+
+6. The django framework is a open-source python web framework that follows the MVC architecture. Its used as a backend tool
+   to help manage everything and comes with a lot of built in functionality that makes us coders not depressed while coding.
+
+7. Serializer and Model Serializers generally do the same thing the difference is that Model Serializer is designed for
+   serializing and deserializing data that corresponds to a django model.
+   it also has built in validation and automatically generates fields based on the models fields. Another important thing
+   that model serializers do is handle the DB operations for you.
+
+   The regular Serializer doesnt create fields based off the django model or have built in validation system.
+   It also doesnt it serializes data that isn't necessarily tied to our DB model.
+
+   You would typically use Model Serializer when your working with data that's built like django models since it provides
+   lots of convenient functionally for data that's designed like a django model.
+   As for regular serializer you use when you're parsing data that's not django model related. regular serializing works
+   with all types of data.
+
+8. There are three different types of relations in the Django model:
+    One to One - acts similarly to ForeignKeys with unique=True,
+    which lets have 2 fields in separate models that are connect like a person and a profile. each person has one profile.
+
+    Many to One - Same ideas as one to one just here one model has one field that alot of different modals are connected to.
```

Comment:
Worth noting that it's defined with ForeignKey, and that it can be used to create Many-To-One relationship too.